### PR TITLE
Assign test owners to default as a fallback. Pick a random owner if there are multiple options.

### PR DESCRIPTION
**Pull request by rmmh**
See also: kubernetes/kubernetes#29564

### CODE REVIEW

File: mungegithub/mungers/testowner/owner.go [hunk 1]
```diff
@@ -41,18 +42,28 @@ func normalize(name string) string {
...
 }
 
-// TestOwner returns the owner for a test, or the empty string if none is found.
+// TestOwner returns the owner for a test, an owner from $default if present,
+// or else the empty string if none is found.
 func (o *OwnerList) TestOwner(testName string) string {
 	name := normalize(testName)
 	owner, _ := o.mapping[name]
+	if owner == "" {
+		owner, _ = o.mapping["$default"]
```
Comment (fejta) [inline mungegithub/mungers/testowner/owner.go]: I'm not super wild about $default as this looks like a environment variable

**Comment by fejta** [comment]
LGTM

File: mungegithub/mungers/testowner/owner.go [hunk 2]
```diff
@@ -41,18 +42,28 @@ func normalize(name string) string {
...
+	}
+	if strings.Contains(owner, "/") {
```
Comment (fejta) [inline mungegithub/mungers/testowner/owner.go]: Would a comma or tab make more sense here?

File: mungegithub/mungers/testowner/owner.go (hunk 1 above)
Comment (rmmh) [inline mungegithub/mungers/testowner/owner.go]: What would you prefer?

File: mungegithub/mungers/testowner/owner.go (hunk 2 above)
Comment (rmmh) [inline mungegithub/mungers/testowner/owner.go]: It's in a CSV, so commas are right out. The / is more visible/understandable than a tab.

File: mungegithub/mungers/testowner/owner.go (hunk 1 above)
Comment (fejta) [inline mungegithub/mungers/testowner/owner.go]: `DEFAULT` would be clear enough for me

File: mungegithub/mungers/testowner/owner.go (hunk 2 above)
Comment (fejta) [inline mungegithub/mungers/testowner/owner.go]: Makes sense about the comma... Maybe a [semi-colon](https://en.wikipedia.org/wiki/Comma-separated_values#Specification)?

Comment (rmmh) [inline mungegithub/mungers/testowner/owner.go]: `/` is more commonly used to separate items than `;`.

File: mungegithub/mungers/testowner/owner.go [hunk 3]
```diff
@@ -41,18 +42,28 @@ func normalize(name string) string {
 // OwnerList uses a map to get owners for a given test name.
 type OwnerList struct {
 	mapping map[string]string
+	rng     *rand.Rand
 }
 
-// TestOwner returns the owner for a test, or the empty string if none is found.
+// TestOwner returns the owner for a test, an owner from $default if present,
```
Comment (ixdy) [inline mungegithub/mungers/testowner/owner.go]: comment needs updating

**Pull request merged**